wagtail/admin/action_menu.py

```diff
@@ -248,10 +249,16 @@ def __init__(self, request, **kwargs):
         page = self.context.get('page')
         if page:
             task = page.current_workflow_task
+            is_final_task = page.current_workflow_state and page.current_workflow_state.is_at_final_task
             if task:
                 actions = task.get_actions(page, request.user)
-                workflow_menu_items = [WorkflowMenuItem(name, label, launch_modal) for name, label, launch_modal in actions]
-                workflow_menu_items = [item for item in workflow_menu_items if item.is_shown(self.request, self.context)]
+                workflow_menu_items = []
+                for name, label, launch_modal in actions:
+                    if name == "approve" and is_final_task:
+                        label = format_lazy(_("{label} and Publish"), label=label)
```

Comment:
We should use old-style string interpolation here - Transifex doesn't recognise this style of format string, and will not guard against translators trying to translate the word "label" (which will cause an error for that one language).

Reply:
There are a bunch of these in register_workflow_log_actions too, so will do that en masse in a follow-up PR